tests/test_dense.py

```diff
@@ -1,101 +1,174 @@
-import unittest
 import sys
-from mock_data.dense_matrix_19_genes_100k_cells_txt.gene_models_0 import gene_models
+import unittest
+from unittest.mock import patch
 
 sys.path.append("../ingest")
-from dense import Dense
+from expression_files.dense_ingestor import DenseIngestor
+from mock_data.dense_matrix_19_genes_100k_cells_txt.gene_models_0 import gene_models
 
 
 class TestDense(unittest.TestCase):
-    def test_validate_gene_keyword_false(self):
-        "Validates validate_gene_keyword() returns false correctly"
-        expression_matrix = Dense(
-            '../tests/data/expression_matrix_bad_missing_keyword.txt',
-            '5d276a50421aa9117c982845',
-            '5dd5ae25421aa910a723a337',
-            tracer=None,
-        )
-        self.assertFalse(expression_matrix.validate_gene_keyword())
-        expression_matrix.close()
-
-    def test_validate_gene_keyword_true(self):
-        "Validates validate_gene_keyword() returns true correctly"
-        expression_matrix = Dense(
-            '../tests/data/dense_expression_matrix.txt',
-            '5d276a50421aa9117c982845',
-            '5dd5ae25421aa910a723a337',
-            tracer=None,
-        )
-        self.assertTrue(expression_matrix.validate_gene_keyword())
-        expression_matrix.close()
-
-    def test_validate_unique_header_false(self):
-        "Validates validate_unique_header() returns false correctly"
-        expression_matrix = Dense(
-            '../tests/data/expression_matrix_non_unique.txt',
-            '5d276a50421aa9117c982845',
-            '5dd5ae25421aa910a723a337',
-            tracer=None,
+    # TODO add test method after SCP-2635 implemented
+
+    def test_process_header(self):
+        header = ["one", "two", '"three', "'four"]
+        actual_header = DenseIngestor.process_header(header)
+        self.assertEqual(["one", "two", "three", "four"], actual_header)
+
+    def test_header_has_valid_values(self):
+        header = ["one", "two", '"three', "'four"]
+        empty_value = ["", "two"]
+        space_value = ["   ", "two"]
+        nan_value = ["nan", "two"]
+
+        self.assertTrue(DenseIngestor.header_has_valid_values(header))
+        self.assertFalse(DenseIngestor.header_has_valid_values(empty_value))
+        self.assertFalse(DenseIngestor.header_has_valid_values(space_value))
+        self.assertFalse(DenseIngestor.header_has_valid_values(nan_value))
+
+    def test_filter_expression_scores(self):
+        scores = ["BRCA1", 4, 0, 3, "0", None, "", "   ", "nan", "NaN", "Nan", "NAN"]
+        cells = ["foo", "foo2", "foo3", "foo4", "foo5", "foo6", "foo7"]
+        invalid_scores = ["BRCA1", 4, 0, 3, "0", None, "T"]
```

Comment:
moving this declaration of invalid_scores down to right above where it's used will make the test easier to read

Reply:
c5a6ac3